src/redux/books/booksSlice.js

```diff
@@ -1,42 +1,78 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+const BASE_URL = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/LIwynBVImvIPWv4fs65F/books';
 
 const initialState = {
-  bookstore: [
-    {
-      item_id: 'item1',
-      title: 'The Great Gatsby',
-      author: 'John Smith',
-      category: 'Fiction',
-    },
-    {
-      item_id: 'item2',
-      title: 'Anna Karenina',
-      author: 'Leo Tolstoy',
-      category: 'Fiction',
-    },
-    {
-      item_id: 'item3',
-      title: 'The Selfish Gene',
-      author: 'Richard Dawkins',
-      category: 'Nonfiction',
-    }],
+  bookstore: [],
+  status: 'idle',
+  error: null,
+  createStatus: '',
+  deleteStatus: '',
 };
+export const fetchBooks = createAsyncThunk('books/fetchBooks', async () => {
+  try {
+    const response = await axios.get(BASE_URL);
+    return response.data;
+  } catch (error) {
+    return error.message;
+  }
+});
+export const postBooks = createAsyncThunk('books/postBooks', async (initialbooks) => {
+  try {
+    const response = await axios.post(BASE_URL, initialbooks);
+    return response.data;
+  } catch (error) {
+    return error.message;
+  }
+});
+
+export const deleteBooks = createAsyncThunk('books/deleteBooks', async (bookid) => {
+  try {
+    const response = await axios.delete(BASE_URL + bookid);
+    return response.data;
+  } catch (error) {
+    return error.message;
+  }
+});
```

Comment:
- [ ] On adding books, there is no change in the browser. Kindly fix that, please. If a user adds a book, the book should be added to the API and also displayed in the booklist but that is not the case for your application. Kindly fix this, please.